sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/BuilderHelperTests.java

```diff
@@ -0,0 +1,568 @@
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.FixedDelayOptions;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.RetryOptions;
+import com.azure.core.test.http.MockHttpResponse;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.DateTimeRfc1123;
+import com.azure.core.util.Header;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import com.azure.storage.common.policy.RequestRetryOptions;
+import com.azure.storage.common.policy.RetryPolicyType;
+import com.azure.storage.file.share.implementation.util.BuilderHelper;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class BuilderHelperTests {
+
+    private static final StorageSharedKeyCredential CREDENTIALS = new StorageSharedKeyCredential("accountName", "accountKey");
+    private static final String ENDPOINT = "https://account.file.core.windows.net/";
+    private static final RequestRetryOptions REQUEST_RETRY_OPTIONS = new RequestRetryOptions(RetryPolicyType.FIXED, 2, 2, 1000L, 4000L, null);
+    private static final RetryOptions CORE_RETRY_OPTIONS = new RetryOptions(new FixedDelayOptions(1, Duration.ofSeconds(2)));
+
+    private static HttpRequest request(String url) throws MalformedURLException {
+        return new HttpRequest(
+            HttpMethod.HEAD,
+            new URL(url),
+            new HttpHeaders().set(HttpHeaderName.CONTENT_LENGTH, "0"),
+            Flux.empty());
+    }
+
+    /**
+     * Tests that a new date will be applied to every retry when using the default pipeline builder.
+     */
+    @Test
+    public void freshDateAppliedOnRetry() throws MalformedURLException {
+        HttpClient httpClient = new FreshDateTestClient();
+        HttpLogOptions httpLogOptions = BuilderHelper.getDefaultHttpLogOptions();
+        ClientOptions clientOptions = new ClientOptions();
+
+        BuilderHelper.buildPipeline(CREDENTIALS, null, null, null, null, REQUEST_RETRY_OPTIONS, null, httpLogOptions,
```

Comment:
This test needs to use `StepVerifier`, right now this will subscribe to the asynchronous request and won't wait for it to complete and will exit execution before any testing is performed. This will need to be fixed in other tests using this same pattern.